File: src/main/java/com/qtech/bubbles/bubble/AccelerateBubble.java
package com.qtech.bubbles.bubble;

import com.qtech.bubbles.common.ResourceLocation;
import com.qtech.bubbles.common.entity.Entity;
import com.qtech.bubbles.core.utils.categories.ColorUtils;
import com.qtech.bubbles.entity.BubbleEntity;
import com.qtech.bubbles.entity.player.PlayerEntity;
import org.apache.commons.lang.math.DoubleRange;
import org.apache.commons.lang.math.IntRange;

public class AccelerateBubble extends AbstractBubble {
    public AccelerateBubble() {
        colors = ColorUtils.parseColorString("#00003f,#00007f,#0000af,#0000ff");
        setRegistryName(ResourceLocation.fromString("qbubbles:accelerate_bubble"));

//        BubbleInit.BUBBLES.add(this);

//        setPriority(244000);
        setPriority(2440000);
        setRadius(new IntRange(25, 54));
        setSpeed(new DoubleRange(6.0, 28.0));
        setDefense(1.0f);
        setAttack(0.001f);
        setScore(1);
        setHardness(0.7d);
    }

    @Override
    public void onCollision(BubbleEntity source, Entity target) {
        if (target instanceof PlayerEntity) {
            PlayerEntity player = (PlayerEntity) target;

            // Calculate Velocity X and Y.
            double accelerateX = 0;
            double accelerateY = 0;
            if (player.isMotionEnabled()) {
                accelerateX += Math.cos(Math.toRadians(player.getRotation())) * 0.625d;
                accelerateY += Math.sin(Math.toRadians(player.getRotation())) * 0.625d;
            }

            // Set velocity X and Y.
            player.setAccelerateX(player.getAccelerateX() + accelerateX);
            player.setAccelerateY(player.getAccelerateY() + accelerateY);
        }
    }
}

File: src/main/kotlin/com/qtech/bubbles/bubble/AccelerateBubble.kt
package com.qtech.bubbles.bubble

import com.qtech.bubbles.common.ResourceLocation
import com.qtech.bubbles.entity.Entity
import com.qtech.bubbles.core.utils.categories.ColorUtils
import com.qtech.bubbles.entity.BubbleEntity
import com.qtech.bubbles.entity.player.PlayerEntity
import org.apache.commons.lang.math.DoubleRange
import org.apache.commons.lang.math.IntRange

class AccelerateBubble : AbstractBubble() {
    override fun onCollision(source: BubbleEntity, target: Entity) {
        if (target is PlayerEntity) {
            val player = target

            // Calculate Velocity X and Y.
            var accelerateX = 0.0
            var accelerateY = 0.0
            if (player.isMotionEnabled) {
                accelerateX += Math.cos(Math.toRadians(player.rotation)) * 0.625
                accelerateY += Math.sin(Math.toRadians(player.rotation)) * 0.625
            }

            // Set velocity X and Y.
            player.accelerateX = player.accelerateX + accelerateX
            player.accelerateY = player.accelerateY + accelerateY
        }
    }

    init {
        colors = ColorUtils.parseColorString("#00003f,#00007f,#0000af,#0000ff")
        registryName = ResourceLocation.fromString("bubbleblaster:accelerate_bubble")

//        BubbleInit.BUBBLES.add(this);

//        setPriority(244000);
        priority = 2440000.0
        radius = IntRange(25, 54)
        speed = DoubleRange(6.0, 28.0)
        defense = 0.1f
        attack = 0.001f
        score = 1f
        hardness = 0.7
    }
}